lua/plenary/path.lua

```diff
@@ -428,8 +439,26 @@ function Path:_split()
   return vim.split(self:absolute(), self._sep)
 end
 
+local _get_parent = (function()
+  local formatted = string.format('^(.+)%s[^%s]+', path.sep, path.sep)
+  return function(abs_path)
+    return abs_path:match(formatted)
+  end
+end)()
+
+function Path:parent()
+  return _get_parent(self:absolute()) or path.root
+end
+
 function Path:parents()
-  return self:absolute():match(string.format('^(.+)%s[^%s]+', self._sep, self._sep))
+  local results = {}
+  local cur = self:absolute()
+  repeat
+    cur = _get_parent(cur)
+    table.insert(results, cur)
+  until not cur
+  table.insert(results, path.root)
```

Comment:
Here is what i mean why we need a dynamic solution for windows. Basically have a path.root for returns the root to the given path.

Imagine `neovim` running in `C:\repos` (whatever) `path.root` would be `C:\` because `vim.loop.cwd() == C:\repos` but the user can still do `Telescope file_browser cwd=F:\\archive` (e.g.) and the prompt thingy would then show `C:\\archive` you see my point. So we would need `path.root(input)` and on linux we just always return `/` and for windows we do `sub(1,1) .. ':\\`